runner/logs.go

```diff
@@ -140,7 +140,9 @@ func downloadLogsForServer(server *state.Server, globalCollectionOpts state.Coll
 	}
 
 	transientLogState := state.TransientLogState{CollectedAt: time.Now()}
-	defer transientLogState.Cleanup()
+	defer func() {
```

Comment:
I wonder if we should add a note here, that's saying something like:

```golang
// We wrap the function call in an anonymous function to avoid a copy of the stack-allocated
// struct occurring at this point - causing the log file list pointed to by the state to be incorrect
// (as it gets replaced in a subsequent function call, causing the pointer to change)
```

(I've also briefly considered whether we can fix this by using a pointer, but that seems more complicated?)

Reply:
Updated the code to fix this with pointer receivers instead. Confirmed by running an instrumented build on our staging collector.